diff --git a/core/packages.el b/core/packages.el
index 8b66296d..0a24997d 100644
--- a/core/packages.el
+++ b/core/packages.el
@@ -40,3 +40,4 @@
 ;; core-keybinds.el
 (package! which-key)
 (package! hydra)
+(package! company-ycmd)
diff --git a/modules/completion/company/config.el b/modules/completion/company/config.el
index 7bc2154c..bde5d775 100644
--- a/modules/completion/company/config.el
+++ b/modules/completion/company/config.el
@@ -27,8 +27,9 @@ MODES should be one major-mode symbol or a list of them."
   :commands (company-mode global-company-mode company-complete
              company-complete-common company-manual-begin company-grab-line)
   :init
-  (setq company-idle-delay nil
+  (setq company-idle-delay 0.2
         company-tooltip-limit 14
+  :config
         company-dabbrev-downcase nil
         company-dabbrev-ignore-case nil
         company-dabbrev-code-other-buffers t
diff --git a/modules/lang/javascript/config.el b/modules/lang/javascript/config.el
index 50b51543..a24c9d7e 100644
--- a/modules/lang/javascript/config.el
+++ b/modules/lang/javascript/config.el
@@ -225,3 +225,18 @@
   :files "package.json"
   :add-hooks (+javascript|add-node-modules-path))
 
+  :on-enter
+  (when (make-local-variable 'exec-path)
+    (push (doom-project-expand "node_modules/.bin")
+          exec-path)))
+
+
+;;
+;; Tools
+;;
+
+(def-project-mode! +javascript-eslintd-fix-mode
+  :add-hooks (eslintd-fix-mode))
+           (setq ycmd-server-command '("python" "/Users/baljeetkumar/code_repositories/YouCompleteMe/third_party/ycmd/run_ycmd.sh"))
+           (setq ycmd-force-semantic-completion t)
+           (add-hook 'js2-mode-hook 'ycmd-mode)
diff --git a/modules/lang/org/packages.el b/modules/lang/org/packages.el
index dc3c1cb2..498c20d3 100644
--- a/modules/lang/org/packages.el
+++ b/modules/lang/org/packages.el
@@ -5,6 +5,13 @@
 (package! org-plus-contrib)
 
 (package! org-bullets :recipe (:fetcher github :repo "Kaligule/org-bullets"))
+;(when (version< emacs-version "26.1")
+  ;; We want org 9.1.x, but the org packaged with Emacs 25.x and under is 8.x.
+  ;; The only secure (and reasonably trustworthy) source for this is via
+  ;; emacsmirror. Emacs 26+ comes with Org 9.1.4.
+ ; (package! org-plus-contrib
+  ;  :recipe (:fetcher github :repo "emacsmirror/org" :files (:defaults "contrib/lisp/*.el"))))
+
 (package! toc-org)
 
 (when (featurep! :feature evil)
diff --git a/modules/tools/term/config.el b/modules/tools/term/config.el
index adb1d470..84395e95 100644
--- a/modules/tools/term/config.el
+++ b/modules/tools/term/config.el
@@ -6,3 +6,9 @@
   (setq multi-term-program (getenv "SHELL")
         multi-term-dedicated-window-height 20
         multi-term-switch-after-close 'PREVIOUS))
+(let ((path (shell-command-to-string ". ~/.zshrc; echo -n $PATH")))
+  (setenv "PATH" path)
+  (setq exec-path 
+        (append
+         (split-string-and-unquote path ":")
+         exec-path)))
